sources/packages/backend/apps/api/src/route-controllers/student-appeal/student-appeal.students.controller.ts

```diff
@@ -117,9 +118,20 @@ export class StudentAppealStudentsController extends BaseController {
     let dryRunSubmissionResults: DryRunSubmissionResult[] = [];
     try {
       const dryRunPromise: Promise<DryRunSubmissionResult>[] =
-        payload.studentAppealRequests.map((appeal) =>
-          this.formService.dryRunSubmission(appeal.formName, appeal.formData),
-        );
+        payload.studentAppealRequests.map((appeal) => {
+          // Check if the form has any non submission data(data which is not persisted but required for validation)
+          // which needs to be populated at the server side for dry run submission.
+          if (appeal.formData instanceof StudentAppealNonSubmissionData) {
```

Comment:
The idea of using the `instanceof` is to enforce the type, hence the program year can be assigned as below.
```ts
if (appeal.formData instanceof StudentAppealNonSubmissionData) {
  appeal.formData.programYear = application.programYear.programYear;
}
```